### DISCUSSION TITLE
Timeouts using anycable-rails

### DISCUSSION THREAD

**Discussion by ajporterfield** [anycable/anycable#196] [General]
I'm running into trouble migrating a basic Rails app from the out-of-the-box Action Cable setup to AnyCable.  I thought I would reach out b/c there must be something I'm over looking.  Here are some details that I'm hoping will be helpful.

It's a Rails 7 app I created this morning.  I'm using Ruby 2.7.6p219.  The only thing I did was use the scaffold generator to create a new Message resource.  Here's what my messages_channel.rb looks like.

    class MessagesChannel < ApplicationCable::Channel
      def subscribed
        puts "subscribed"
        stream_from "my_messages"
      end

      def unsubscribed
        # Any cleanup needed when channel is unsubscribed
      end

      def my_method(data)
        puts data
      end
    end

And here's what messages_channel.js looks like.

    import consumer from "channels/consumer"

    consumer.subscriptions.create("MessagesChannel", {
      connected() {
        console.log("connected");
      },

      disconnected() {
        console.log("disconnected");
      },

      received(data) {
        console.log(`received data: ${data}`);
      }
    });

I'm using the Rails console to broadcast (ex: `ActionCable.server.broadcast("my_messages", { test: "payload" })`).  I'm checking the browser console and the Rails server log to verify my console.log's and puts are firing as expected.

I first got this working with the out of the box Rails Action Cable setup.  I did switch the adapter to Redis so I could broadcast from the console.  I also verified I could connect, subscribe, and call my_method from Postman (external app).

Next, I followed the [anycable-rails README](https://github.com/anycable/anycable-rails) and switched over to using AnyCable.  I ran the setup generator which was very helpful.  Unfortunately, I'm seeing a lot of `Socket connection to 'ws://localhost:8080/cable' failed:` errors in the browser console.  It does connect after several automatic retries, but usually gets disconnected after a short time.   It's pretty much unusable.  
<img width="432" alt="Screen Shot 2023-04-14 at 2 15 07 PM" src="https://user-images.githubusercontent.com/2551544/232130982-0011300a-f022-4767-a47a-1d6a49f13296.png">

I also can't connect from Postman anymore.  It tries to connect for a few seconds then returns the following error.
<img width="738" alt="Screen Shot 2023-04-14 at 2 52 15 PM" src="https://user-images.githubusercontent.com/2551544/232131906-c93f5b61-8083-4bac-9129-618b9f8f6da8.png">

I'm running on a Mac (M1).  All the dependencies installed and start up without any issue.  I don't see any errors coming from the terminal (I'm using hivemind to start them using the Procfile created by the setup generator).

Do you have any thoughts as to why I'm running into trouble getting/staying connected?  Thanks for your help!

**Comment by ajporterfield** [comment]
Quick update: A co-worker of mine went through the setup with the same Ruby/Rails version using this same Rails app repo and didn't run into of these connection issues.  So, that's promising.

**Comment by palkan** [comment]
Can you please provide some details? How do you launch the app and its component? What do you see in the `anycable-go` output?

**Comment by ajporterfield** [comment]
Thanks for the reply @palkan.  [Here's the repo](https://github.com/ajporterfield/anycable).  I'm starting things by running these commands each in their own terminal tab (also included in the above repo's README).

    bundle exec anycable
    anycable-go --host=localhost --port=8080
    bundle exec rails s

I have also double checked that Redis is running.  Here's the anycable-go output when it starts up.  I'm not seeing any additional anycable-go output during the disconnects/reconnects.

<img width="871" alt="Screen Shot 2023-04-16 at 9 27 50 AM" src="https://user-images.githubusercontent.com/2551544/232315030-925022b6-76d1-44ed-bc7a-af9beb6a3985.png">

**Reply by palkan** [reply]
Can you try running `anycable-go` with debug logs, i.e., `anycable-go --host=localhost --port=8080 -debug`?
